tokio/src/runtime/task/join.rs

```diff
@@ -203,6 +203,37 @@ impl<T> JoinHandle<T> {
         }
     }
 
+    /// Checks if the task associated with the handle has finished(whether completed or cancelled).
+    /// This function does not block.
+    ///
+    /// ```rust
+    /// use tokio::time;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///    let handle1 = tokio::spawn(async {
+    ///        // do some stuff here
```

Comment:
Adding a call to `time::pause` ensures that this test doesn't actually take a full second to run. The `#` hides it from the documentation.
```suggestion
    /// #[tokio::main]
    /// async fn main() {
    /// #  tokio::time::pause();
    ///    let handle1 = tokio::spawn(async {
    ///        // do some stuff here
```